cspatients/tests.py

```diff
@@ -218,33 +193,160 @@ def test_insufficient_post_method(self):
             template_name="cspatients/view.html"
         )
+
+# Util Tests
+
+
+class ViewAllContextTest(TestCase):
+
+    def setUp(self):
+        delete_whole_table()
+        self.patient_one_data = {
+            "name": "Jane Doe",
+            "patient_id": "XXXXX",
+            "age": 20,
+        }
+
+        self.patient_two_data = {
+            "name": "Mary Mary",
+            "patient_id": "YYYYY",
+            "age": 23,
+            "urgency": 1
+        }
+
+    def test_context_when_no_patients(self):
+        delete_whole_table()
+        self.assertFalse(view_all_context())
+
+    def test_context_when_patiententrys(self):
+        delete_whole_table()
+        patient_one = PatientSerializer(data=self.patient_one_data)
+        patiententry_one = PatientEntrySerializer(data=self.patient_one_data)
+        if patient_one.is_valid():
+            patient_one.save()
+            if patiententry_one.is_valid():
+                patiententry_one.save()
+        patient_two = PatientSerializer(data=self.patient_two_data)
+        patiententry_two = PatientEntrySerializer(data=self.patient_two_data)
+        if patient_two.is_valid():
+            patient_two.save()
+            if patiententry_two.is_valid():
+                patiententry_two.save()
+
+        # Check that view_all_context doesn't return False
+        self.assertTrue(view_all_context)
+
+        # Check that it returns two objects
+        self.assertEqual(len(view_all_context()), 2)
+
+        # Check that the results are sorted by the urgency
+        self.assertEqual(view_all_context()[0].patient_id.name, "Mary Mary")
+
+
+class SaveModelTest(TestCase):
+
+    def setUp(self):
+        delete_whole_table()
+        self.patient_one_data = {
+            "name": "Jane Doe",
+            "patient_id": "XXXXX",
+            "age": 20,
+        }
+
+        self.patient_two_data = {
+            "name": "Mary Mary",
+            "age": 23,
+            "urgency": 1
+        }
+
+    def test_saves_when_given_sufficient_data(self):
+        patiententry = save_model(self.patient_one_data)
+
+        # Must return the saved patient entry
+        self.assertIsNotNone(patiententry)
+        # Check that the save persisted in the database
+        self.assertTrue(PatientEntry.objects.get(patient_id="XXXXX"))
+
+    def test_nothing_saved_with_insufficient_data(self):
+        patientryentry = save_model(self.patient_two_data)
+
+        # Must return None
+        self.assertIsNone(patientryentry)
+
+        # Check that none of the information is in the database
+        with self.assertRaises(Patient.DoesNotExist):
+            Patient.objects.get(name="Mary Mary")
+
+
+class SaveModelChangesTest(TestCase):
+
+    def setUp(self):
+        delete_whole_table()
+        self.patient_one_data = {
+            "name": "Jane Doe",
+            "patient_id": "XXXXX",
+            "age": 20,
+        }
+        save_model(self.patient_one_data)
```

Comment:
It's probably better practice to create this patient using standard Django methods (with `Patient.object.create(..`), rather than relying on code that you've written. Like this, if there's a bug in your `save_model` method it will break these tests as well.